src/tests/system/tests/test_sudo.py

```diff
@@ -556,3 +556,35 @@ def test_sudo__local_users_negative_cache(client: Client, provider: LDAP, sssd_s
 
     result = client.tools.tshark(["-r", "/tmp/sssd.pcap", "-V", "-2", "-R", "ldap.filter"])
     assert "uid=user-1" not in result.stdout
+
+
+@pytest.mark.importance("critical")
+@pytest.mark.authorization
+@pytest.mark.topology(KnownTopologyGroup.AnyProvider)
+def test_sudo__defaults_rule(client: Client, provider: GenericProvider):
+    """
+    :title: defaults sudo rule behavior with no authentication
+    :setup:
+        1. Create user "user-1"
+        2. Create a sudorule named defaults with option '!authenticate'
+        3. Create a sudorule named allow-user-1 to allow user-1 to run all commands on all hosts with password
+           authentication required
```

Comment:
If you want to test this, shouldn't you add `option="authenticate"` to the second rule?

Reply:
For this test case, I'm dropping the explicit 'authentication required' part from this scenario. At this test case, I want to test the scenario where 'authentication' part is not mentioned. The 'cn=defaults' entry will contain '!authenticate'. 
Expectation here would be if a sudo rule does not have a specific option(here 'authenticate') defined then cn=defaults should be referred to see if that option is defined. In current scenario, cn=defaults has  an option defined for authentication.

Reply:
We don't have that test currently, so we should add it. Can you add it to this PR? All tests inherit undefined, so we can skip that one.

Reply:
I've added a test case for cn=defaults set !authentication with sudorule mandatory auth. 
Now two test cases for options in cn=defaults
`cn=defaults          sudorule`
`!authentication      undefined auth`
`!authentication      mandatory auth`